ksql-engine/src/main/java/io/confluent/ksql/function/UdfLoader.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.function;
+
+import com.google.common.base.Preconditions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import io.confluent.ksql.function.udf.PluggableUdf;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.SchemaUtil;
+import io.github.lukehutch.fastclasspathscanner.FastClasspathScanner;
+
+public class UdfLoader {
+
+  private static final Logger logger = LoggerFactory.getLogger(UdfLoader.class);
+
+  private final MetaStore metaStore;
+  private final File pluginDir;
+  private final ClassLoader parentClassLoader;
+  private final Predicate<String> blacklist;
+  private final UdfCompiler compiler;
+  private final boolean loadCustomerUdfs;
+
+
+  public UdfLoader(final MetaStore metaStore,
+                   final File pluginDir,
+                   final ClassLoader parentClassLoader,
+                   final Predicate<String> blacklist,
+                   final UdfCompiler compiler,
+                   final boolean loadCustomerUdfs) {
+    this.metaStore = Objects.requireNonNull(metaStore, "metaStore can't be null");
+    this.pluginDir = Objects.requireNonNull(pluginDir, "pluginDir can't be null");
+    this.parentClassLoader = Objects.requireNonNull(parentClassLoader,
+        "parentClassLoader can't be null");
+    this.blacklist = Objects.requireNonNull(blacklist, "blacklist can't be null");
+    this.compiler = Objects.requireNonNull(compiler, "compiler can't be null");
+    this.loadCustomerUdfs = loadCustomerUdfs;
+  }
+
+  public void load() {
+    // load udfs packaged as part of ksql first
+    loadUdfs(parentClassLoader);
```

Comment:
Is this where the `Substring` class (which is updated in this PR) is going to be loaded and compiled?

Reply:
yes